### PR TITLE
Added logout button

### PR DESCRIPTION
**Pull request by BlackConure**
Simple update to the boilerplate which allows returning back to the main page after a login name has been locally stored.

### CODE REVIEW

**Comment by BlackConure** [comment]
Rats!  Forgot to put in the tests.   Added in this latest push.

**Comment by pronebird** [comment]
Great work! 🎉 

Travis errors because of `read-config` package that does not properly transpile for node 6.

Since tests run in `electron-mocha` which spins off Electron, it really makes no point to test on different versions of Node.js.

Would you mind updating `.travis.yml` and removing everything except node 8?

```diff
node_js:
  - '8'
-  - '7'
-  - '6'
```

File: app/components/LoggedIn.js [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
+  };
+
+  handleLogout = () => {
+    this.props.onLogout({
+      username: '',
+      loggedIn: false
+    });
+  }
+
+
```
Comment (jschr) [inline app/components/LoggedIn.js]: Can we remove the additional newline?

File: app/containers/LoggedInPage.js [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+      dispatch(push('/'));
+    }
+  };
 };
 
 export default connect(mapStateToProps, mapDispatchToProps)(LoggedIn);
+
+
+
+
```
Comment (jschr) [inline app/containers/LoggedInPage.js]: Here too

Comment (pronebird) [reply app/containers/LoggedInPage.js]: Good spot. Btw I've been using `prettier` for a while now and we never had to bother about code style ever again. May make sense to add `yarn format` to re-format all the code, that also includes trailing newlines etc..

Comment (jschr) [reply app/containers/LoggedInPage.js]: Yeah definitely agreed! I use `prettier` in all my projects now. I think it's definitely worthwhile to add.

**Comment by BlackConure** [comment]
Sorry about the blank lines...I have been using Beautify (but it is manual) and am looking at prettier now.   I have it installed in VSCode and in the project as a part of the pre-commit.   

However, in VSCode it is messing up with JSX.     Check out how it formatted the render:

  render() {
    return ( < div >
      <
      h2 > Logged in as {
        this.props.user.username
      } < /h2> <button onClick = {this.handleLogout} > Logout </button > < /div>
    );
  }

Which of the 10 million Prettier plug-ins are you using?

I haven't looked at the pre-commit to see if that is working yet.

**Comment by pronebird** [comment]
I use prettier from terminal to be fair, i.e 

```
yarn add prettier --global
```

then run as `prettier --write 'app/**/*.js' 'test/**/*.js'`

My current config `.prettierrc.yml`:

```
# .prettierrc.yml
# see: https://prettier.io/docs/en/options.html
printWidth: 100
tabWidth: 2
semi: true
useTabs: false
singleQuote: true
trailingComma: all
bracketSpacing: true
jsxBracketSameLine: true
arrowParens: always
proseWrap: always
```

**Comment by BlackConure** [comment]
Crap!   Nevermind....Beautify was still active and they were conflicting......ok.....I will get a new commit here shortly.

**Comment by jschr** [comment]
@BlackConure don't worry about adding prettier in this PR. We've opened up an issue which we'll track separately: https://github.com/jschr/electron-react-redux-boilerplate/issues/56

Just remove the lines manually and update the travis config per pronebirds suggestion and we're all set!

**Comment by pronebird** [comment]
What @jschr said, no point to bother with exact style, we'll reformat everything later.

**Comment by BlackConure** [comment]
Ok will do...few mins.

**Comment by BlackConure** [comment]
Sorry....looks like prettier default was double quotes and now I just found another place it needs to be corrected.    Re-submitting.

**Comment by BlackConure** [comment]
FYI...if you are using VSCode, you might want to look at the add-in Prettier Now instead of Esben Petersen's version.   It seems to be better about JSX, and it also didn't break long lines (nor put double quotes in as a default).

**Comment by pronebird** [comment]
Yeah, prettier uses double quote in JSX and I think I read that JSX has some issues with single quotes when it's impossible to escape `'` or something like that. We'll probably relax ESLint settings when prettier is added to this boilerplate.

**Comment by jschr** [comment]
Thanks @BlackConure!

**Pull request merged**